src/main/java/org/marketplace/controllers/AdvertisementManagementController.java

```diff
@@ -1,22 +1,68 @@
 package org.marketplace.controllers;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 
 import org.marketplace.models.Advertisement;
 import org.marketplace.models.Category;
 import org.marketplace.services.AdvertisementManagementService;
 import org.marketplace.services.CategoryManagementService;
 import org.springframework.web.bind.annotation.*;
 
+import java.util.List;
+
 @RestController
-@RequestMapping("/advertisement")
+@RequestMapping("/api/advertisement")
 public class AdvertisementManagementController {
+
     private final AdvertisementManagementService advertisementManagementService;
     public AdvertisementManagementController(AdvertisementManagementService advertisementManagementService) {
         this.advertisementManagementService = advertisementManagementService;
     }
 
     @PostMapping("/add")
-    public Advertisement requestAddOffer(@RequestBody Advertisement advertisement) {
+    public Advertisement requestAddAdvertisement(@RequestBody Advertisement advertisement) {
         return advertisementManagementService.addAdvertisement(advertisement);
     }
+
+    @DeleteMapping("/{id}")
+    public void requestDeleteAdvertisement(@PathVariable Long id) {
+        advertisementManagementService.deleteAdvertisement(id);
+    }
+
+    @PutMapping("/{id}")
+    public Advertisement requestUpdateAdvertisement(@RequestBody Advertisement advertisement) {
+        return advertisementManagementService.updateAdvertisement(advertisement);
+    }
+
+    @GetMapping("/{id}")
+    public Advertisement requestGetAdvertisement(@PathVariable Long id) {
+        return advertisementManagementService.getAdvertisementById(id);
+    }
+
+    @GetMapping("/all")
+    public List<Advertisement> requestGetAllAdvertisements() {
+        return advertisementManagementService.getAllAdvertisements();
+    }
+
+    @GetMapping("/category/{id}")
+    public List<Advertisement> requestGetAdvertisementsByCategory(@PathVariable Long id) {
+        return advertisementManagementService.getAdvertisementsByCategory(id);
+    }
+
+    @GetMapping("/user/{id}")
+    public List<Advertisement> requestGetAdvertisementsByUser(@PathVariable Long id) {
+        return advertisementManagementService.getAdvertisementsByUser(id);
+    }
+
+    /*
+
+    additional methods that can be implemented in the future:
+
+    - deleting all user ads
+    - deleting all category ads
+    - getting a sum of categories
+
+     */
```

Comment:
just very minor thing
i could suggest to not add //TODO into a production code, might be better to keep as less comments as possible,
it's fine to do that for documentation purpose, like

    /**
     * Retrieves a greeting message.
     *
     * @param name the name of the person
     * @return a greeting message
     */
    @GetMapping("/greet")
    public String greet(@RequestParam("name") String name) {
        return "Hello, " + name + "!";
    }
    ```
    or use a swagger that generates html with doc, like
    
    ```
    import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import io.swagger.annotations.ApiParam;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@Api(tags = "GreetingController", description = "Operations pertaining to greetings")
public class GreetingController {

    @GetMapping("/greet")
    @ApiOperation(value = "Get a greeting message", notes = "Provide a name to get a personalized greeting message")
    public String greet(@ApiParam(value = "Name of the person", required = true) @RequestParam String name) {
        return "Hello, " + name + "!";
    }
}
it's also totally fine to not use documetation at all.